### DISCUSSION TITLE
Adding new components to livespice

### DISCUSSION THREAD

**Discussion by PastaCodex** [dsharlet/LiveSPICE#220] [Q&A] [answered]
I was looking to use a very specific opamp (LM386) in a guitar pedal I want to emulate. Does anyone know how to add these components or what fields on the components I need to edit to make this work?

**Comment by dsharlet** [comment, marked as answer]
Here are the steps to doing this:
- Find a datasheet for the component. In this case, this looks like the right one? https://www.ti.com/lit/ds/symlink/lm386.pdf
- On page 5, the various characteristics tell you *roughly* what you need to know. It's not always straightforward to map things.
- Using this table in the datasheet, it is possible to populate the parameters of an `OpAmp` component in LiveSPICE:
  - The input resistance `Rin` is given in the table on page 5, `50k`
  - The output resistance `Rout` is a little tricky. I think it's given in the output power. Unfortunately it depends on the power supply voltage(?). It's low enough that it probably doesn't matter though.
  - `Aol` is the open loop gain. It looks like this is given as 26 or 46 dB, depending on how it's used. This is quite low for an op-amp, but it might be correct? That translates to a gain of ~20 or ~200.
  - `GBP` is the unity gain bandwidth. Just eyeballing things, it looks like ~4 MHz

It's been a while since I've done this, I might have gotten part of this grossly wrong. Getting some of these quantities more accurately would require doing a bit of math on the datasheet values.

**Reply by PastaCodex** [reply]
Thanks for pointing me in the right direction. This is really helpful. Once I get a chance to play with this a little more I'll update on if I got it to work.
